glib/src/types.rs

```diff
@@ -351,6 +351,10 @@ builtin!(f32, F32);
 builtin!(f64, F64);
 builtin!(str, STRING);
 builtin!(String, STRING);
+builtin!(std::ffi::OsString, STRING);
+builtin!(std::ffi::OsStr, STRING);
+builtin!(std::path::PathBuf, STRING);
+builtin!(std::path::Path, STRING);
```

Comment:
It's really sad that GLib has annotations for distinguishing `utf8` from `filename` strings but at the type system level (`GType`) it's all a string.

Fortunately it doesn't matter too much because a) who uses non-UTF8 locales on UNIXy systems nowadays and b) filenames are always UTF-8 in GLib on Windows.